js/index.js

```diff
@@ -1,23 +1,32 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
 
-  //... your code goes here
+  const price = parseFloat(product.querySelector(".price span").innerHTML);
+  const quantity = product.querySelector(".quantity input").valueAsNumber;
+  
+  const subtotalNum = price * quantity;
+
+  product.querySelector('.subtotal span').innerHTML = subtotalNum;
+
+  return subtotalNum;
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
+  const availableItems = document.getElementsByClassName('product');
+  
 
   // ITERATION 2
-  //... your code goes here
+  let total = 0;
+
+  for (item of availableItems) {
+    total += updateSubtotal(item);
+  }
 
   // ITERATION 3
-  //... your code goes here
+  document.querySelector('#total-value span').innerHTML = total;
 }
 
 // ITERATION 4
```

Comment:
Good work, I don't see any issues with your code/solution

I'll add a working solution for iteration 4 and 5 so you can check the logic for it in your own time
```
// ITERATION 4
function removeProduct(event) {
  const target = event.currentTarget;
  console.log('The target in remove is:', target);
  // ...
  const row = target.parentNode.parentNode;
  //   console.log('row: ', row);
  const parent = row.parentNode;
  //   console.log('parent: ', parent);
  parent.removeChild(row);

  calculateAll();
}

// ITERATION 5

function createProduct() {
  const createRow = document.querySelector('.create-product');
  let newProdNameInput = createRow.querySelector('input');
  let newProdNameValue = newProdNameInput.value;
  let newProdPriceInput = createRow.querySelector("input[type='number']");
  let newProdPriceValue = Number(newProdPriceInput.valueAsNumber).toFixed(2);

  const newTableRow = document.createElement('tr');
  newTableRow.className = 'product';
  newTableRow.innerHTML = `
    <td class="name">
      <span>${newProdNameValue}</span>
    </td>
    <td class="price">$<span>${newProdPriceValue}</span></td>
    <td class="quantity">
      <input type="number" value="0" min="0" placeholder="Quantity" />
    </td>
    <td class="subtotal">$<span>0</span></td>
    <td class="action">
      <button class="btn btn-remove">Remove</>
    </td>
  `;

  // get the parent of this newly created row
  const parent = document.querySelector('#cart tbody');

  // append the newly created row to the parent
  parent.appendChild(newTableRow);

  // make sure remove button inherits the same behavior as other remove buttons
  const removeBtn = newTableRow.querySelector('.btn-remove');
  removeBtn.addEventListener('click', removeProduct);

  // clean the fields
  newProdNameInput.value = '';
  newProdPriceInput.value = 0;
}